hathor/transaction/storage/remote_storage.py

```diff
@@ -534,13 +533,11 @@ def get_tx_count(self) -> int:
         return result.count
 
     def get_genesis(self, hash_bytes: bytes) -> Optional['BaseTransaction']:
-        if not self._genesis_cache:
-            self._create_genesis_cache()
+        assert self._genesis_cache is not None
         return self._genesis_cache.get(hash_bytes, None)
 
     def get_all_genesis(self) -> Set['BaseTransaction']:
-        if not self._genesis_cache:
-            self._create_genesis_cache()
+        assert self._genesis_cache is not None
```

Comment:
Isn't `self._genesis_cache` removed?

Reply:
The genesis are being added to `_genesis_cache ` on `_save_or_verify_genesis`.